CRM/Contact/BAO/Relationship.php

```diff
@@ -316,7 +316,7 @@ public static function add(&$params, $ids = array(), $contactId = NULL) {
     $relationship->contact_id_b = $params['contact_id_b'];
     $relationship->contact_id_a = $params['contact_id_a'];
     $relationship->relationship_type_id = $type;
-    $relationship->id = $relationshipId;
+    $relationship->id = $params['id'];
```

Comment:
@pradpnayak I worry that this isn't robust & could be changed at any point, breaking any extensions that rely on it,  since there is no specific contract / commitment to permitting id to be altered. 2 ways to protect it are

1) by using code practices as standardised as possible - in this case perhaps merge in the defaults first & then do per code comment

```
 //@todo this code needs to be updated for the possibility that not all fields are set
    // by using $relationship->copyValues($params);


```

2) unit tests